### PR TITLE
SNOW-1760125: Disable test_read_snowflake_query_cte_with_python_anonymous_sproc test from precommit

### PR DESCRIPTION
**Pull request by sfc-gh-lmukhopadhyay**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-1760125

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
      - [ ] If this test skips Local Testing mode, I'm requesting review from @snowflakedb/local-testing
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency
   - [ ] If this is a new feature/behavior, I'm adding the Local Testing parity changes.
   - [ ] I acknowledge that I have ensured my changes to be thread-safe. Follow the link for more information: [Thread-safe Developer Guidelines](https://docs.google.com/document/d/162d_i4zZ2AfcGRXojj0jByt8EUq-DrSHPPnTa4QvwbA/edit#bookmark=id.e82u4nekq80k)

3. Please describe how your code solves the related issue.

   Disabling `test_read_snowflake_query_cte_with_python_anonymous_sproc` flaky test from Precommit run until there is a reliable fix

### CODE REVIEW

File: tests/integ/modin/io/test_read_snowflake_query_cte.py [hunk 1]
```diff
@@ -165,7 +164,6 @@ def filterByRole(session: Session, tableName: String, role: String): DataFrame =
     assert_snowpark_pandas_equals_to_pandas_without_dtypecheck(snow_df, native_df)
 
 
-@pytest.mark.modin_sp_precommit
```
Comment (sfc-gh-azhan) [inline tests/integ/modin/io/test_read_snowflake_query_cte.py:168]: can we add a todo to remind us to reenable it?

Comment (sfc-gh-lmukhopadhyay) [reply tests/integ/modin/io/test_read_snowflake_query_cte.py:168]: Yes, added

Comment (sfc-gh-azhan) [reply tests/integ/modin/io/test_read_snowflake_query_cte.py:168]: Can we track with a ticket?

Comment (sfc-gh-lmukhopadhyay) [reply tests/integ/modin/io/test_read_snowflake_query_cte.py:168]: Yes I added [SNOW-1760467](https://snowflakecomputing.atlassian.net/browse/SNOW-1760467) to track this

**Review by sfc-gh-azhan — APPROVED** [review]

**Review by sfc-gh-helmeleegy — APPROVED** [review]
LGTM, thanks!

**Pull request merged**